### PR TITLE
gh-123884 Tee of tee was not producing n independent iterators

### PR DESCRIPTION
**Pull request by rhettinger**
* Eliminates the code path that reused a tee object input as part of the output.
* Updated the rough equivalent recipe to have better fidelity to the actual C code.
* Removed a note about using the tee input. It is generically true of all iterator pass throughs and the note made it seem like a caution specific to tee where a tee-of-a-tee is the exception to the rule.

* Issue: gh-123884

----
📚 Documentation preview 📚: https://cpython-previews--124490.org.readthedocs.build/

### CODE REVIEW

File: Doc/library/itertools.rst [hunk 1]
```diff
@@ -691,25 +691,36 @@ loops that truncate the stream.
...
-                        link[1] = [None, None]
-                    value, link = link
-                    yield value
-            except StopIteration:
-                return
-
-   Once a :func:`tee` has been created, the original *iterable* should not be
-   used anywhere else; otherwise, the *iterable* could get advanced without
-   the tee objects being informed.
+                raise ValueError
```
Comment (ericvsmith) [inline Doc/library/itertools.rst:694]: Is there any particular reason to remove the text for this exception? Not that it matters much, but the C code does retain the text.

File: Lib/test/test_itertools.py [hunk 2]
```diff
@@ -1759,21 +1760,36 @@ def test_tee_recipe(self):
 
         def tee(iterable, n=2):
             if n < 0:
-                raise ValueError('n must be >= 0')
-            iterator = iter(iterable)
-            shared_link = [None, None]
-            return tuple(_tee(iterator, shared_link) for _ in range(n))
+                raise ValueError
```
Comment (ericvsmith) [inline Lib/test/test_itertools.py:1763]: Same comment here about the exception text.

**Review by ericvsmith — APPROVED** [review]
Other than the one noted nit, this all looks good to me. I verified the "roughly equivalent" code, but not the code in test_itertools.py (I figure the test suite will do that job).

File: Doc/library/itertools.rst (hunk 1 above)
Comment (rhettinger) [reply Doc/library/itertools.rst:694]: I thought it was distracting and would take maintenance effort to keep it synced with the C code.  Also, in a pure python equivalent, the surrounding code is sufficiently informative.

**Pull request merged**